okhttp/src/main/java/io/grpc/okhttp/OkHttpTlsUpgrader.java

```diff
@@ -74,6 +78,14 @@ public static SSLSocket upgrade(SSLSocketFactory sslSocketFactory,
           OkHttpProtocolNegotiator.get().negotiate(sslSocket, host, TLS_PROTOCOLS);
       Preconditions.checkState(HTTP2_PROTOCOL_NAME.equals(negotiatedProtocol),
           "Only \"h2\" is supported, but negotiated protocol is %s", negotiatedProtocol);
+
+      SSLSession session = sslSocket.getSession();
+      if (session == null) {
```

Comment:
These code piece was copied from the latest (SSLCertificateSocketFactory)[https://android.googlesource.com/platform/frameworks/base/+/android-6.0.0_r7/core/java/android/net/SSLCertificateSocketFactory.java#200] implementation, but I think you are right, the Java doc claims it won't return null.

Done.